hw4/04-templates.js

```diff
@@ -1,61 +1,137 @@
-const express = require('express');
-const axios = require('axios');
+/** @format */
+
+const express = require("express");
+const axios = require("axios");
 
 const app = express();
 const port = process.env.PORT || 5001;
 
 // Use Pug as the templating engine
-app.set('views', __dirname + '/views');
-app.set('view engine', 'pug');
+app.set("views", __dirname + "/views");
+app.set("view engine", "pug");
 
 // REST Countries URL
-const url = 'https://restcountries.com/v3.1/all';
-
-// Add your code here
-
-app.get('/', (req, res) => {
+const url = "https://restcountries.com/v3.1/all";
+
+async function getCountriesData(url) {
+  try {
+    let countriesResponse = await axios.get(url);
+    if (countriesResponse.data == null) {
+      console.error("No response was returned for the requested url!");
+    } else {
+      return countriesResponse.data;
+    }
+  } catch (error) {
+    console.error("There is some error with the get operation: ", error);
+  }
+}
+let countriesData = getCountriesData(url);
+
+app.get("/", (req, res) => {
```

Comment:
Nice Implementation instead of writing the code again and again calling single function everytime